docs/protocol/core/v3.0.rst

```diff
@@ -52,6 +54,25 @@ or by making a pull request against the
 This document was produced by the `Zarr core development team
 <https://github.com/orgs/zarr-developers/teams/core-devs>`_.
 
+Goal of v3 spec and main difference from v2
+===========================================
+
+Zarr spec v2 was originally designed around local filesystem, but Zarr has
+grown and is now regularly deployed on cloud / object storage. Those kind of
+storage have characteristics, capabilities and usage patterns that can widely
+differ from the assumptions of spec v2. V3 is design to consider online
+stores in particular we want to achieve the following.
+
+ - No assumption that the underlying store have locking ability.
+ - Ability to do concurrent writes with the assumption that writes from clients will be consistent, but non atomic.
+
+
+Unlike Zarr spec version 2, the spec V3 have mainly the following difference:
+  - V3 is a flat key-value store instead of an hierarchical store. Hierarchy is implied. 
```

Comment:
What are the implications of this on the local filesystem implementations? cc: @jni

Reply:
Not much:
 - you can't open a hierarchy at an arbitrary point, you have to open the root. (zarr_open, could walk up the tree until is finds the root, and return a subgroup, that's not an issue, but that's just user facing).
 - meta/ and data/ are in two subtrees, so the storage format is one layer deeper than v2

Also the chink separator is now `/` by default; which has a much bigger impact on the disk layout as now you have a deeper hierarchy, with less files per folder.

Reply:
> Also the chink separator is now / by default; which has a much bigger impact on the disk layout as now you have a deeper hierarchy, with less files per folder.

Ah, that's the crux. This particular sentence makes it sound like the opposite is true, but they are talking about different things. Thanks for clarifying @Carreau! I'm not sure whether it makes sense to add a similar clarification with this bullet point — I'll leave that to people more familiar with this whole ecosystem. =)